### PR TITLE
CONTRIB-003: Update the dependencies and configurations to fix the stylelint warnings.

### PR DESCRIPTION
**Pull request by AlviMurtaza**
- Add dev dependency `postcss-scss` plugin to fix the stylelint customSyntax warning.
- Update the import sass warning and other configurations.

### CODE REVIEW

**Comment by hussainweb** [comment]
@AlviMurtaza, awesome. Does this mean we were missing a dependency all this time? I ask because this looks like a replacement but you have not removed any of the older packages. Do we still need all the dependencies?

File: web/themes/custom/contribtracker/gulp-tasks/config.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ export default {
         },
       ],
       failOnError: process.env.CI === 'true',
+      debug: true,
```
Comment (hussainweb) [inline web/themes/custom/contribtracker/gulp-tasks/config.js:32]: Is this accidental?

File: web/themes/custom/contribtracker/gulp-tasks/scss.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import autoprefixer from 'autoprefixer';
 import postcss from 'gulp-postcss';
-import dartSass from 'sass'
+import * as dartSass from 'sass'
```
Comment (hussainweb) [inline web/themes/custom/contribtracker/gulp-tasks/scss.js:3]: The error pointed only to this as a problem. Is the `postcss-scss` dependency related to this?

**Review by hussainweb — COMMENTED** [review]
@AlviMurtaza, awesome. Does this mean we were missing a dependency all this time? I ask because this looks like a replacement but you have not removed any of the older packages. Do we still need all the dependencies?

File: web/themes/custom/contribtracker/gulp-tasks/scss.js (hunk 2 above)
Comment (AlviMurtaza) [reply web/themes/custom/contribtracker/gulp-tasks/scss.js:3]: No this was also a warning displaying when running `gulp`
Screenshot 2024-05-20 at 5 52 27 PM

File: web/themes/custom/contribtracker/gulp-tasks/config.js (hunk 1 above)
Comment (AlviMurtaza) [reply web/themes/custom/contribtracker/gulp-tasks/config.js]: Yes, removed.

File: web/themes/custom/contribtracker/gulp-tasks/scss.js (hunk 2 above)
Comment (hussainweb) [reply web/themes/custom/contribtracker/gulp-tasks/scss.js:3]: Yes, that is what I mean. This is the change that will fix the warning. Is the postcss change related to this?

**Comment by AlviMurtaza** [comment]
> @AlviMurtaza, awesome. Does this mean we were missing a dependency all this time? I ask because this looks like a replacement but you have not removed any of the older packages. Do we still need all the dependencies?

Yes `customSyntax` seems to be introduced after `stylelint v14.0.0`, and since we upgraded the versions, this was a new requirement, the customSyntax package depends upon what we are [trying to compile]. (https://github.com/postcss/postcss#syntaxes). Other packages are still required.

File: web/themes/custom/contribtracker/gulp-tasks/scss.js (hunk 2 above)
Comment (AlviMurtaza) [reply web/themes/custom/contribtracker/gulp-tasks/scss.js:3]: yes

Comment (hussainweb) [reply web/themes/custom/contribtracker/gulp-tasks/scss.js:3]: So, you're fixing two different issues in this PR, right? Let's avoid that in the future. Create as many PRs as you need.

**Review by hussainweb — APPROVED** [review]
@AlviMurtaza, merging for now but please remember to separate PRs for different changes.

**Comment by AlviMurtaza** [comment]
> @AlviMurtaza, merging for now but please remember to separate PRs for different changes.

Sure thing 👍

**Pull request merged**